### ISSUE TITLE
Concern of current ICS20 interface breaking Evmos ICS20 precompliles

### ISSUE THREAD

**Issue by womensrights** [cosmos/solidity-ibc-eureka#63]
The current [ICS20 interface](https://github.com/cosmos/solidity-ibc-eureka/blob/main/src/interfaces/IICS20Transfer.sol) would break the [existing Evmos contracts](https://github.com/evmos/evmos/blob/main/precompiles/ics20/ICS20I.sol).

Additional feedback: 

The ERC20 interfaces also are not complete because they are missing key functionality such as:
- metadata
- Allowance
- permit and permit2

**Comment by srdtrk** [comment]
These contracts are not meant to be deployed on evmos or be used in the same way. What type of investigation needs to be performed here?

**Comment by womensrights** [comment]
This is true but it would be worth following up with the evmos team what their concern is

**Comment by womensrights** [comment]
I think we should implement the [metadata interface](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20Metadata) as it is widely used, permit2 [is tracked here](https://github.com/cosmos/solidity-ibc-eureka/issues/162) and has allowance already within the scope

**Comment by gjermundgaraba** [comment]
The metadata interface is already included in the IERC20 interface we use, and permit2 will be implemented as mentioned.

**Comment by gjermundgaraba** [comment]
Given that, I will close this issue for now, until any other concerns are surfaced.
